Jenkinsfile

```diff
@@ -0,0 +1,687 @@
+            echo "Clone using default branch and repository."
+            checkout scm
+        } else {
+            echo "Clone using ${OPENJCEPLUS_BRANCH} from ${OPENJCEPLUS_REPO}"
+            git branch: "${OPENJCEPLUS_BRANCH}", url: "${OPENJCEPLUS_REPO}"
+        }
+    }
+}
+
+/*
+ * Get the appropriate test flag.
+ *
+ * The user might have requested that test are not
+ * run or asked for a specific test to
+ * be executed.
+ *
+ * Even if that's not the case, in some
+ * platforms the OpenJCEPlusFIPS provider is not
+ * available and thus the FIPS-related tests should
+ * not be executed.
+ */
+def getTestFlag(hardware, software) {
+    // User requested that tests not be executed.
+    if (EXECUTE_TESTS == "false") {
+        return " -DskipTests"
+    }
+
+    // User requested execution of a specific test.
+    def specificTest = SPECIFIC_TEST
+    if (specificTest != "") {
+        return " -Dtest=${specificTest}"
+    }
+
+    // Based on platform, decide whether FIPS tests should be run.
+    if ((software == "mac") || ((software == "linux") && (hardware == "aarch64"))) {
+        echo "Only non-FIPS tests can be run in ${hardware}_${software}"
+        return " -Dtest=ibm.jceplus.junit.openjceplus.TestAll," +
+                       "ibm.jceplus.junit.TestMemStressAll," +
+                       "ibm.jceplus.junit.TestMultithread," +
+                       "ibm.jceplus.junit.openjceplus.integration.TestAll"
+    } else {
+        echo "All tests (both FIPS and non-FIPS) can be run in ${hardware}_${software}"
+        return "";
+    }
+}
+
+/*
+ * Export the appropriate environment variables
+ * and run the requested maven commands.
+ */
+def runOpenJCEPlus(command, software) {
+    dir("openjceplus/OpenJCEPlus") {
+        def additional_exports = ""
+        if (software == "aix") {
+            additional_exports = "export LIBPATH=$WORKSPACE/openjceplus/OCK/:$WORKSPACE/openjceplus/OCK/jgsk_sdk;"
+        }
+
+        def additional_envars = ADDITIONAL_ENVARS
+        if (additional_envars != "") {
+            for (envar in additional_envars.split(",")) {
+                additional_exports += " export ${envar.trim()};"
+            }
+            
+        }
+
+        def java_home = "export JAVA_HOME=$WORKSPACE/java/jdk;"
+        def gskit_home = "export GSKIT_HOME=$WORKSPACE/openjceplus/OCK/jgsk_sdk;"
+        def environment = "export PATH=$WORKSPACE/apache-maven-3.9.6/bin:\$PATH;"
+        def ock_path = "$WORKSPACE/openjceplus/OCK/"
+        if (software == "windows") {
+            def workspace_unix_style = "$WORKSPACE".replace("\\", "/").replace("C:", "/cygdrive/c")
+            environment = """export PATH=${workspace_unix_style}/apache-maven-3.9.6/bin:/cygdrive/c/Program\\ Files\\ \\(x86\\)/Windows\\ Kits/10/bin/10.0.19041.0/x64/:/cygdrive/c/Program\\ Files/Microsoft\\ Visual\\ Studio/2022/Professional/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/:\$PATH;\\
```

Comment:
For this version of VC tools (e.g., 14.31.31103), will the environment always have the same version across all the Windows build platforms?

Reply:
Hello for now yes all the systems have VC tools in this location. Over time we may have the change this if we decide to use another version. There are no plans at this time to change this.

Reply:
Ok, got it.